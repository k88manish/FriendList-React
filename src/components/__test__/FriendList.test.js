import React from "react";
import { configure } from "enzyme";
import { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-15";
import FriendList from "../FriendList";

configure({ adapter: new Adapter() });

const mockPageLoad = jest.fn();
const propsData = {
  friends: [{ name: "manish", id: 1, gender: "Male" }, { name: "mark", id: 2, gender: "Female" }],
  actions: {
    starFriend: jest.fn()
  }
};

describe("FriendList Test Suits", () => {
  it("it should render correctly", () => {
    const componentWrapper = shallow(<FriendList {...propsData} />);
    expect(componentWrapper).toMatchSnapshot();
  });

  it("List should be having 2 items", () => {
    const componentWrapper = shallow(<FriendList {...propsData} />);
    expect(componentWrapper.find("FriendListItem").length).toEqual(2);
  });
});
