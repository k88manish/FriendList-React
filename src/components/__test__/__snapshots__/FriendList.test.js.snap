exports[`FriendList Test Suits it should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FriendList
    actions={
        Object {
            "starFriend": [Function],
          }
    }
    friends={
        Array [
            Object {
              "gender": "Male",
              "id": 1,
              "name": "manish",
            },
            Object {
              "gender": "Female",
              "id": 2,
              "name": "mark",
            },
          ]
    } />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <FriendListItem
          gender="Male"
          id={1}
          name="manish"
          starFriend={[Function]}
          starred={undefined} />,
        <FriendListItem
          gender="Female"
          id={2}
          name="mark"
          starFriend={[Function]}
          starred={undefined} />,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "gender": "Male",
          "id": 1,
          "name": "manish",
          "starFriend": [Function],
          "starred": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "class",
        "props": Object {
          "gender": "Female",
          "id": 2,
          "name": "mark",
          "starFriend": [Function],
          "starred": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <FriendListItem
            gender="Male"
            id={1}
            name="manish"
            starFriend={[Function]}
            starred={undefined} />,
          <FriendListItem
            gender="Female"
            id={2}
            name="mark"
            starFriend={[Function]}
            starred={undefined} />,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "gender": "Male",
            "id": 1,
            "name": "manish",
            "starFriend": [Function],
            "starred": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "class",
          "props": Object {
            "gender": "Female",
            "id": 2,
            "name": "mark",
            "starFriend": [Function],
            "starred": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
